docs/secret-rotation.md

```diff
@@ -0,0 +1,254 @@
+
+By following the steps outlined in **Create/Delete Credentials**, the option to delete application credentials can be found.
+
+Do note that in the scenario where the secret has been compromised, this should be the first step completed; this is because doing so invalidates the credential posing the security threat and prevents any operations using those credentials from being authenticated. Apart from this, and no longer needing to rename **credentials.yaml** to **old-credentials.yaml**, all other steps are the same.
+
+<a id="rotating-compromised-secrets-link"></a>
+###	**Rotating Compromised Secrets**
+As mentioned above, an application credential which is a security concern should be deleted before any other step; after which a new application credential, which will replace it, can be created.
+
+| Don’t forget to download the clouds.yaml file after creating the new application credential\! |
+| :---- |
+
+Due to having deleted the old application credentials there is no need to rename the **credentials.yaml** file and can instead be directly replaced with the template, plus the contents of the new application credential’s **clouds.yaml**.
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <MAKE SURE THIS MATCHES “cloudCredentialsSecretName:” IN configmap.yaml>
+  namespace: capi-self
+  annotations:
+    # Allow the sealed secret controller to take over this secret after bootstrapping
+    sealedsecrets.bitnami.com/managed: "true"
+stringData:
+  clouds.yaml: |
+    clouds:
+      openstack:
+        auth:
+          auth_url: ...
+          application_credential_id: ...
+          application_credential_secret: ...
+        interface: ...
+        identity_api_version: 3
+        auth_type: ...
+```
+
+Then, much like the previous secret rotation, use Kubeseal to encrypt the new secret. This time round, the Kubeseal command below will seal the secret and overwrite **credentials.yaml** rather than create an intermediate **encrypted-creds.yaml** file.
+
+| kubeseal \--kubeconfig kubeconfig \--format yaml \--controller-name sealed-secrets \--controller-namespace sealed-secrets-system \--secret-file credentials.yaml  \--sealed-secret-file credentials.yaml |
+| :---- |
+
+Again, this file will then be added, committed and pushed to a preferably new, GitHub branch which can then be reviewed and merged into the branch which FluxCD is tracking for changes. After which, once per interval, FluxCD will compare the current configuration with the branch on GitHub, and upon noticing any differences will attempt to update the current cluster’s configuration to reflect the changes.
+
+<a id="validate-configuration-link"></a>
+# **Validate Configuration**
+
+It is always a good idea to test your deployments for any potential risks or errors, which can lead to more issues further down the line. Therefore, it is worthwhile to validate the Kubernetes cluster with a diagnostic tool.
+
+<a id="sonobuoy-link"></a>
+###	**Sonobuoy**
```

Comment:
Does the Sonobuoy output contain any confirmation that the new cloud credentials worked? E.g. by provisioning a `PVC` which creates a Cinder volume or a k8s `Service` of type `LoadBalancer` which triggers the creation of an OpenStack LB? This was the main motivation for including Sonobuoy in the secret rotation process.

Reply:
Discussed last week and how doesn't share if it was a success or not.